### PR TITLE
Update odp-dpdk/dpdk pktio start function

### PR DESCRIPTION
**Pull request by bogdanPricope**
Set RX drop enable option and cleanup the TX code

### CODE REVIEW

File: platform/linux-dpdk/odp_packet_dpdk.c [hunk 1]
```diff
@@ -324,6 +324,41 @@ static int close_pkt_dpdk(pktio_entry_t *pktio_entry ODP_UNUSED)
 	return 0;
 }
 
+static odp_bool_t tx_checksum(pktio_entry_t *pktio_entry, uint32_t *txq_flags)
+{
+	uint64_t ip_ena  = pktio_entry->s.config.pktout.bit.ipv4_chksum_ena;
+	uint64_t udp_ena = pktio_entry->s.config.pktout.bit.udp_chksum_ena;
+	uint64_t tcp_ena = pktio_entry->s.config.pktout.bit.tcp_chksum_ena;
+	uint64_t sctp_ena = pktio_entry->s.config.pktout.bit.sctp_chksum_ena;
```
Comment (muvarov) [inline platform/linux-dpdk/odp_packet_dpdk.c]: why uint64_t? Has to be the same size as register or odp_bool_t.

File: platform/linux-dpdk/odp_packet_dpdk.c [hunk 2]
```diff
@@ -324,6 +324,41 @@ static int close_pkt_dpdk(pktio_entry_t *pktio_entry ODP_UNUSED)
 	return 0;
 }
 
+static odp_bool_t tx_checksum(pktio_entry_t *pktio_entry, uint32_t *txq_flags)
+{
+	uint64_t ip_ena  = pktio_entry->s.config.pktout.bit.ipv4_chksum_ena;
```
Comment (muvarov) [inline platform/linux-dpdk/odp_packet_dpdk.c]: should pktio lock be locked here?

Comment (bogdanPricope) [reply platform/linux-dpdk/odp_packet_dpdk.c]: Entry locks are locked - this is called from start() function of the dpdk pktio that is executed while locks are locked (inside odp_pktio_start())

Comment (muvarov) [reply platform/linux-dpdk/odp_packet_dpdk.c]: than it's ok.

File: platform/linux-dpdk/odp_packet_dpdk.c (hunk 1 above)
Comment (bogdanPricope) [reply platform/linux-dpdk/odp_packet_dpdk.c]: "In all implementations, the default integer type for a bit field is unsigned."

I guess unsigned should do.

File: platform/linux-dpdk/odp_packet_dpdk.c [hunk 3]
```diff
@@ -324,6 +324,41 @@ static int close_pkt_dpdk(pktio_entry_t *pktio_entry ODP_UNUSED)
 	return 0;
 }
 
+static odp_bool_t tx_checksum(pktio_entry_t *pktio_entry, uint32_t *txq_flags)
+{
+	unsigned ip_ena  = pktio_entry->s.config.pktout.bit.ipv4_chksum_ena;
+	unsigned udp_ena = pktio_entry->s.config.pktout.bit.udp_chksum_ena;
+	unsigned tcp_ena = pktio_entry->s.config.pktout.bit.tcp_chksum_ena;
+	unsigned sctp_ena = pktio_entry->s.config.pktout.bit.sctp_chksum_ena;
```
Comment (muvarov) [inline platform/linux-dpdk/odp_packet_dpdk.c]: I'm ok with this change. But would following is more compact:
``

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**